themes/default/content/blog/tour-pulumi-equinix-provider/index.md

```diff
@@ -0,0 +1,163 @@
+
+{{% notes type="info" %}}
+**Note:** The Equinix provider effectively replaces the now-deprecated [Equinix Metal provider](https://www.pulumi.com/registry/packages/equinix-metal/). To migrate your Pulumi resources to the new provider, use the [`pulumi import`](https://www.pulumi.com/docs/cli/commands/pulumi_import/) command.
+{{% /notes %}}
+
+## Creating a Kubernetes cluster on Equinix Metal
+
+In order to demonstrate the power and utility of Pulumi and the Equinix provider, Equinix Labs has produced a codebase that [creates a Kubernetes cluster on Equinix Metal](https://github.com/equinix-labs/pulumi-equinix-kubernetes-cluster/). The codebase is available in both [TypeScript](https://github.com/equinix-labs/pulumi-equinix-kubernetes-cluster/tree/main/nodejs) and [Python](https://github.com/equinix-labs/pulumi-equinix-kubernetes-cluster/tree/main/python).
+
+The codebase gives an excellent example of one of Pulumi's most compelling features: the ability to manage and orchestrate many different kinds of resources in real programming languages with a single tool. In addition to the Equinix provider which is used to manage the bare metal compute resources, the codebase also uses the following providers:
+
+1. [Cloud-init](https://www.pulumi.com/registry/packages/cloudinit/), to run initialization scripts that install the necessary services to run Kubernetes on the bare metal instances once they are spun up. To get details on what the Cloud-Init scripts are doing on the control plane and worker nodes respectively, see [Kubernetes the Hard Way](https://github.com/kelseyhightower/kubernetes-the-hard-way).
+1. [TLS](https://www.pulumi.com/registry/packages/tls/), to manage cryptographic resources that allow nodes to join the cluster.
+1. [Command](https://www.pulumi.com/registry/packages/command/), which is used to synchronize cluster joining operations and to read the kubeconfig from the control plane.
+1. [Random](https://www.pulumi.com/registry/packages/random/), which is used to generate a token for nodes to join the control plane.
+
+Before running the Pulumi program, you'll need to [configure your Equinix credentials](https://www.pulumi.com/registry/packages/equinix/installation-configuration/#credentials). Once that's done, you can run the following commands to deploy the cluster:
+
+{{% chooser language "typescript,python" / %}}
+
+{{% choosable language typescript %}}
+
+```bash
+cd nodejs
+npm i
+pulumi up
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```bash
+cd python
+python -m venv venv
+source venv/bin/activate
+pip install -r requirements.txt
+pulumi up
+```
+
+{{% /choosable %}}
+
+When the commands complete, you'll have a working cluster. From there, you can write the cluster's Kubeconfig to a file and set it as your default file:
```

Comment:
It would also be great to highlight how fast it works

```suggestion
After executing the commands, the process will require approximately 5 minutes to finish, resulting in a fully operational cluster. From there, you can write the cluster's Kubeconfig to a file and set it as your default file:
```

Reply:
Will be added to updated PR (see below).